vllm/entrypoints/openai/protocol.py

```diff
@@ -550,7 +551,7 @@ class ChatCompletionResponseStreamChoice(OpenAIBaseModel):
     index: int
     delta: DeltaMessage
     logprobs: Optional[ChatCompletionLogProbs] = None
-    finish_reason: Optional[Literal["stop", "length", "tool_calls"]] = None
```

Comment:
Not directly related to removing this check but since finish_reason like 'abort' is not part of the Open AI spec wondering if we it makes sense to do a validation of the returned SequenceStatus.get_finished_reason against the allowed values and if it is not one of them then set the finish_reason to None? E.g. In L:332 in serving_chat.py check first check output.finish_reason and if it is not one of the allowed values then in ChatCompletionResponseStreamChoice response set it to None.